src/Microsoft.Diagnostics.Runtime/src/DataTargets/DataTarget.cs

```diff
@@ -237,88 +235,102 @@ public IEnumerable<ModuleInfo> EnumerateModules()
             RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? new LinuxFunctions() :
             RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? new MacOSFunctions() :
             (PlatformFunctions)new WindowsFunctions();
-
+        
         /// <summary>
-        /// Loads a dump file.  Currently supported formats are ELF coredump and Windows Minidump formats.
+        /// Loads a dump stream. Currently supported formats are ELF coredump and Windows Minidump formats.
         /// </summary>
-        /// <param name="filePath">The path to the dump file.</param>
-        /// <param name="cacheOptions">The caching options to use </param>
-        /// <returns>A <see cref="DataTarget"/> for the given dump file.</returns>
-        public static DataTarget LoadDump(string filePath, CacheOptions? cacheOptions = null)
+        /// <param name="displayName">The name of this DataTarget, might be used in exceptions.</param>
+        /// <param name="stream">The stream that should be used. Ownership will be transferred to the DataTarget.</param>
+        /// <param name="cacheOptions">The caching options to use. (Only used for FileStreams)</param>
+        /// <returns>A <see cref="DataTarget"/> for the given dump.</returns>
+        public static DataTarget LoadDumpFromStream(string displayName, Stream stream, CacheOptions? cacheOptions = null)
         {
-            if (filePath is null)
-                throw new ArgumentNullException(nameof(filePath));
-            else if (!File.Exists(filePath))
-                throw new FileNotFoundException($"Could not open dump file '{filePath}'.", filePath);
-
-            cacheOptions ??= new CacheOptions();
-
-            (FileStream stream, DumpFileFormat format) = OpenDump(filePath);
             try
             {
+                if (displayName is null)
+                    throw new ArgumentNullException(nameof(displayName));
+                if (stream is null)
+                    throw new ArgumentNullException(nameof(stream));
+
+                cacheOptions ??= new CacheOptions();
+
+                DumpFileFormat format = ReadFileFormat(stream);
```

Comment:
The entire library assumes the dump is at the beginning of the stream.  You should either set `stream.Position = 0` or require the user set the position to 0 and throw an exception if `stream.Position != 0`.